### DISCUSSION TITLE
CurseForge and game-versions

### DISCUSSION THREAD

**Discussion by encode42** [Kira-NT/mc-publish#68] [Q&A] [answered]
I'm trying to publish a mod pack to CurseForge, which does not have any files that values can be inferred from (fabric.mod.json, etc.) As such, I'm providing the `game-versions` field myself. This works as expected when publishing to Modrinth, however, the same is not true for CurseForge.

When the action runs, the following error appears:
```
Error: HttpError: 400 (Bad Request, {"errorCode":1002,"errorMessage":"Error in field `metadata`:\n* Array item count 0 is less than minimum count of 1. Path \u0027gameVersions\u0027, line 1, position 381."})
```

[Failing run](https://github.com/Encode42/Natureal/actions/runs/5116059953/jobs/9197848001)

Is this user error? Am I misunderstanding how this field works?

**Comment by Kira-NT** [comment, marked as answer]
Nope, I do not see any problems with your configuration, especially considering the fact that it works fine with Modrinth.

I'm moving this discussion to issues. See #69, please.

---

Update: 

The issue arose from my initial assumption that mod packs necessitated a loader to be specified on CurseForge, just like with individual mods. However, I was surprised to learn that the opposite is true - CurseForge explicitly prohibits the specification of loaders for mod packs. In light of this discovery, the action's behavior has been adjusted accordingly. Everything should be operational now.
